<?xml version="1.0"?>

<launch>
    
    
    <param name="/use_sim_time" value="true" /> 


  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find prius_gazebo)/worlds/intersection.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="false"/>
    <arg name="recording" value="false"/>
    <arg name="debug" value="false"/>
    <arg name="verbose" value="true"/>
    <!-- more default parameters can be changed here -->
  </include>


<!-- Generation of racecar and detectors: lidar, camera, zed(?) -->
<!-- Begin of prius0, moving along y-dir -->
  <group ns="prius0">  
    <param name="tf_prefix" value="prius0" />  
    <include file="$(find prius_gazebo)/launch/includes/prius.xml">
      <arg name="init_pose" value="-x -1 -y 25 -z 0 -Y -1.57" />
      <arg name="robot_name" value="prius0" />
    </include>
  </group>


<!-- Begin of prius1, moving along x-dir -->
  <group ns="prius1">  
    <param name="tf_prefix" value="prius1" />  
    <include file="$(find prius_gazebo)/launch/includes/prius.xml">
      <arg name="init_pose" value="-x -25 -y -1 -z 0 -Y 0" />
      <arg name="robot_name" value="prius1" />
    </include>
  </group>



<!-- Generation of moving obstacle 
  <include file="$(find od_gazebo)/launch/includes/obstacles.xml" if="$(arg use_od)"/>
-->

</launch>
